### PR TITLE
Supply air temperature setting native unit

### LINKED ISSUE #23: Supply air temperature setting not showing units
### Home Assistant version

2026.3.4

### Integration version

1.0.0-beta.4

### System Health details

_No response_

### Checklist

- [x] I have enabled debug logging for my installation.
- [x] I have filled out the issue template to the best of my ability.
- [x] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).
- [x] This issue is not a duplicate issue of any [previous issues](https://github.com/janmilinds/ha-airfi/issues?q=is%3Aissue+label%3A%22Bug%22+)..

### Describe the issue

When having supply air temperature setting in a dashboard tile, no temperature unit is shown. Also, the current state is displaying the setting temperature whereas it would be more beneficial to show the current supply air temperature from the sensor value.

### Reproduction steps

1. Setup integration with air handling unit
2. Add supply air temperature setting to a dashboard
3. Note that
    - no unit of measurement is displayed in widget
    - displayed current state is the temperature setting, not the supply air sensor reading

### Debug logs

```text
n/a
```

### Diagnostics dump

_No response_

### PR DESCRIPTION
**Pull request by janmilinds**
## Description

This pull request introduces enhancements to the temperature number entity in the Airfi integration, focusing on exposing additional attributes and improving unit handling. It also updates icon mappings for better visual clarity and adds corresponding unit tests to ensure correct behavior.

**Enhancements to temperature number entity:**

* Added a `native_unit_of_measurement` property to `AirfiTemperatureNumber`, ensuring the entity always reports its unit as Celsius (`UnitOfTemperature.CELSIUS`). [[1]](diffhunk://#diff-6e82060a83d76f5e5dd87b2f58e9acce632e93a6283c45862225c83d39b239d0R6-R12) [[2]](diffhunk://#diff-6e82060a83d76f5e5dd87b2f58e9acce632e93a6283c45862225c83d39b239d0R60-R82)
* Introduced an `extra_state_attributes` property to expose the current measured supply-air temperature as an attribute, making this data available to Lovelace UI components.

**Testing improvements:**

* Added unit tests to verify the correct reporting of the native unit and the presence (or absence) of the `supply_air_temperature` attribute, including edge cases where input registers may be missing. [[1]](diffhunk://#diff-c41deeddbf8e64a80c592be1f71c4472872ed9c4e978f64ad987cb40232847ebR10-R11) [[2]](diffhunk://#diff-c41deeddbf8e64a80c592be1f71c4472872ed9c4e978f64ad987cb40232847ebR38-R73)

**Icon mapping updates:**

* Updated several icon mappings in `icons.json`:
  * Added an explicit `"off"` state for the `fan` icon.
  * Corrected the icon for the `"unavailable"` state of `firmware_version`.
  * Updated defaults for `sauna_function` and `boosted_cooling` to use more appropriate icons.

## Related Issue

Resolves #23 

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Code quality improvements

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) document
- [x] My code follows the code style of this project (run `script/lint`)
- [x] I have updated the documentation accordingly
- [x] I have updated the translations if needed

### CODE REVIEW

File: custom_components/airfi/icons.json [hunk 1]
```diff
@@ -73,16 +74,10 @@
...
-        }
+        "default": "mdi:heat-wave"
       },
       "boosted_cooling": {
-        "default": "mdi:fan-off",
-        "state": {
-          "on": "mdi:fan-plus"
-        }
+        "default": "mdi:fan-plus"
       }
```
[the remark this answers is not in the document]
Comment (janmilinds) [reply custom_components/airfi/icons.json:81]: Intended change

File: custom_components/airfi/icons.json [hunk 2]
```diff
@@ -12,7 +12,8 @@
       "fan": {
         "default": "mdi:fan-off",
         "state": {
-          "on": "mdi:fan"
+          "on": "mdi:fan",
+          "off": "mdi:briefcase"
```
[the remark this answers is not in the document]
Comment (janmilinds) [reply custom_components/airfi/icons.json:16]: Intended change for ”away” mode
